File: python/1049.py
textoA = input()
textoB = input()
textoC = input()

if (textoA  == "vertebrado"):
    if (textoB == "ave"):
        if (textoC == "carnivoro"):
            print("aguia")
        else:
            print("pomba")
    else:
        if (textoC == "onivoro"):
            print("homem")
        else:
            print("vaca")  
else:
    if (textoB == "inseto"):
        if (textoC == "hematofago"):
            print("pulga")
        else:
            print("lagarta")
    else:
        if (textoC == "hematofago"):
            print("sanguessuga")
        else:
            print("minhoca")

File: 1049.c
#include <stdio.h>
#include <string.h>

int main(){
    char textoA[12];
    char textoB[8];
    char textoC[10];

    scanf("%s", textoA);
    scanf("%s", textoB);
    scanf("%s", textoC);

    if (strcmp(textoA, "vertebrado") == 0){
        if (strcmp(textoB,"ave") == 0){
            if (strcmp(textoC,"carnivoro")==0){
                printf("aguia\n");
            }else{
                printf("pomba\n");
            }
        }else{
            if (strcmp(textoC,"onivoro")==0){
                printf("homem\n");
            }else{
                printf("vaca\n");
            }  
        }
    }else{
        if (strcmp(textoB,"inseto")==0){
            if (strcmp(textoC,"hematofago")==0){
                printf("pulga\n");
            }else{
                printf("lagarta\n");
            }
        }else{
            if (strcmp(textoC,"hematofago")==0){
                printf("sanguessuga\n");
            }else{
                printf("minhoca\n");
            }  
        }
    }
    
}